src/compiler/parser.ts

```diff
@@ -5390,4 +5420,596 @@ module ts {
+                // This is so that /** * @type */ doesn't parse.
+                let canParseTag = true;
+                let seenAsterisk = true;
+
+                for (pos = start + 3; !error && pos < end; ) {
+                    let ch = content.charCodeAt(pos);
+
+                    if (ch === CharacterCodes.at && canParseTag) {
+                        // Don't update 'pos' in this block.  that will be handled inside the 
+                        // parseTag function.
+
+                        parseTag();
+                        
+                        // Once we parse out a tag, we cannot keep parsing out tags on this line.
+                        canParseTag = false;
+                        continue;
+                    }
+
+                    pos++;
+                    if (isLineBreak(ch)) {
+                        // After a line break, we can parse a tag, and we haven't seen as asterisk
+                        // on the next line yet.
+                        canParseTag = true;
+                        seenAsterisk = false;
+                        continue;
+                    }
+
+                    if (isWhiteSpace(ch)) {
+                        // Whitespace doesn't affect any of our parsing.
+                        continue;
+                    }
+
+                    // Ignore the first asterisk on a line.
+                    if (ch === CharacterCodes.asterisk) {
+                        if (seenAsterisk) {
+                            // If we've already seen an asterisk, then we can no longer parse a tag
+                            // on this line.
+                            canParseTag = false;
+                        }
+                        seenAsterisk = true;
+                        continue;
+                    }
+
+                    // Anything else is doc comment text.  We can't do anything with it.  Because it
+                    // wasn't a tag, we can no longer parse a tag on this line until we hit the next
+                    // line break.
+                    canParseTag = false;
+                }
+            }
+        }
+
+        return error ? undefined : createJSDocComment();
+
+        function createJSDocComment(): JSDocComment {
+            if (!returnType && !type && !parameters && !typeParameters) {
+                return undefined;
+            }
+
+            return { returnType, type, parameters, typeParameters };
+        }
+
+        function skipWhitespace(): void {
+            while (pos < end && isWhiteSpace(content.charCodeAt(pos))) {
+                pos++;
+            }
+        }
+
+        function parseTag(): void {
+            Debug.assert(content.charCodeAt(pos) === CharacterCodes.at);
+        
+            // Skip the @ sign.
+            pos++;
+
+            let tagName = scanIdentifier();
+            switch (tagName) {
+                case "param":
+                    return handleParamTag();
+                case "return":

```

Comment:
@returns is a valid synonym of @return according to usejsdoc.org, so I think it should be supported too. The docs seem to even imply that returns is the preferred one. Some IDEs use returns when autogenerating JSDoc comments.

http://usejsdoc.org/tags-returns.html